### PR TITLE
Add more to GitHub workflow, make it support PRs

### PR DESCRIPTION
**Pull request by Jared-Fogle**
Ideally replaces Travis.

Closes #48

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
oh god its the error i dont know how to solve

**Comment by SpaceManiac** [comment]
use `ldd` to find out which dependency is missing and install it as part of the script

**Comment by Jared-Fogle** [comment]
image
image
Ready. @spookydonut was saying something about CI times over 10 minutes, and I believe those also don't run Windows and Linux side by side.

**Comment by spookydonut** [comment]
> 
> 
> image
> image
> Ready. @spookydonut was saying something about CI times over 10 minutes, and I believe those also don't run Windows and Linux side by side.

uncached travis jobs we something like 10 minutes for windows and 14 minutes for linux

**Review by Cyberboss — APPROVED** [review]

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -28,11 +40,33 @@ jobs:
           sudo dpkg --add-architecture i386
           sudo apt-get update
           sudo apt-get install g++-multilib zlib1g-dev:i386 libssl-dev:i386 pkg-config:i386
+          env BYOND_MAJOR=513 BYOND_MINOR=1521 ./scripts/install_byond.sh
```
Comment (ZeWaka) [inline .github/workflows/rust.yml:43]: not latest?

Comment (Jared-Fogle) [reply .github/workflows/rust.yml:43]: I copied this from the old Travis script, not sure if there's an immediately better option.

Comment (ZeWaka) [reply .github/workflows/rust.yml:43]: I mean, the TG dockerfile uses [1533](https://github.com/tgstation/tgstation/blob/master/Dockerfile) at least. 1536 would also work I presume (latest at time of comment, what goon runs).

Comment (Cyberboss) [reply .github/workflows/rust.yml:43]: Either way, i'd set these higher up in a global env section

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Put BYOND version in global `env:` section

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -33,14 +33,16 @@ jobs:
   build-linux:
     runs-on: ubuntu-latest
     env:
+      BYOND_MAJOR: 513
+      BYOND_MINOR: 1521
```
Comment (ZeWaka) [inline .github/workflows/rust.yml:37]: still 🤔

Comment (Cyberboss) [reply .github/workflows/rust.yml:37]: vOv

**Review by Cyberboss — APPROVED** [review]

File: .github/workflows/rust.yml (hunk 2 above)
Comment (ZeWaka) [reply .github/workflows/rust.yml:37]: literally using that version for no reason, even the /tg/ dockerfile is newer...

Comment (Cyberboss) [reply .github/workflows/rust.yml:37]: https://github.com/tgstation/rust-g/issues/56

**Pull request merged**